modules/cluster-ingress/README.md

```diff
@@ -0,0 +1,49 @@
+# Purpose
+The purpose of this module is to deploy kubernetes resources related to ingress for
+the cluster. Along with the ingress we will also deploy out the related SSL cert issuer.
+
+## To implemement
+The Envoy Gateway can secure ingress by verifying JWT. It can be applied to a specific
+target, for example this applies it to all requests going through a `Gateway` called `eg`
+
+```
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: SecurityPolicy
+metadata:
+  name: jwt-example
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: eg
+  jwt:
+    providers:
+    - name: auth0
+      remoteJWKS:
+        uri: https://dev-57n3awu5je6q653y.us.auth0.com/.well-known/jwks.json
+```
```

Comment:
I thought this was a really nifty way to define how the envoy gateway is able to authenticate requests passing through it. There are a number of pre-defined ways shown here: https://gateway.envoyproxy.io/latest/tasks/security/jwt-authentication/ that will allow us to integrate with auth providers.

Another example here for the OIDC Authentication shows how we might be able to do something like stand up a user login screen in front of the cluster using Synapse (Such as how DCA works):
https://gateway.envoyproxy.io/latest/tasks/security/oidc/

Reply:
That's really neat!  I wonder how feasible that would be.